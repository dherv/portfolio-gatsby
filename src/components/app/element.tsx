import React, { FC } from "react";
import { AppHeading } from "../../styled/Heading";
import * as styles from "./element.module.css";

const Element: FC<Props> = ({ title, children }) => {
  return (
    <>
      <AppHeading width={150}>{title}</AppHeading>
      <div className={styles.container}> {children}</div>
    </>
  );
};

interface Props {
  title: string;
}

export default Element;
