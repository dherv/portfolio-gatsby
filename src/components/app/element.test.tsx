import React from "react";
import { render, screen } from "@testing-library/react";
import Element from "./element";
import { sampleProps } from "./element.sample";

test("should display a heading", () => {
  render(<Element {...sampleProps} />);
  expect(screen.getByText("Where I lived")).toBeInTheDocument();
});

test("should display the app heading styled component correctly", () => {
  const { asFragment } = render(<Element {...sampleProps} />);
  expect(asFragment()).toMatchSnapshot();
});
