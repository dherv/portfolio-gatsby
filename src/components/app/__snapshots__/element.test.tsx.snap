// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the app heading styled component correctly 1`] = `
<DocumentFragment>
  <h5
    class="sc-AxiKw dVnzdp"
    width="150"
  >
    Where I lived
  </h5>
  <div
    class="container"
  >
     
    <ul
      class="container"
    >
      <li
        class="text"
      >
        code
      </li>
      <li
        class="text"
      >
        jiujitsu
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
